### ISSUE TITLE
Error 500 on slug field validation for new pages

### ISSUE THREAD

**Issue by simonbru** [infoportugal/wagtail-modeltranslation#264]
An error 500 occurs when *saving/publishing a new page* with a non-unique slug.

This was reported in these issues:
https://github.com/infoportugal/wagtail-modeltranslation/issues/98
https://github.com/infoportugal/wagtail-modeltranslation/issues/112

This issue has been fixed when *saving an existing page*, but for new pages. The issue happens because the code that validates slugs needs the parent page, but unsaved Page instances don't have a parent yet (i.e. `page.get_parent()` returns `None`):
https://github.com/infoportugal/wagtail-modeltranslation/blob/28b304aa487ca89ff65f4cbb6e1bde0b062695f8/wagtail_modeltranslation/patch_wagtailadmin.py#L333

To work around this issue, we override `WagtailAdminPageForm` to validate slugs:
```python
class WagtailFixedAdminPageForm(WagtailAdminPageForm):
    """
    Validate unicity of the slugs in every language. Take fallbacks into account.
    """

    def clean(self):
        cleaned_data = super().clean()

        # We follow the same logic as
        # `wagtail_modeltranslation.patch_wagtailadmin._validate_slugs`,
        # but we make sure that it work with new Page instances.

        siblings = self.parent_page.get_children()
        if self.instance.pk:
            siblings = siblings.not_page(self.instance)

        for code, name in settings.LANGUAGES:
            field_name = build_localized_fieldname("slug", code)
            slug_value = self.cleaned_data.get(field_name, None)
            if slug_value is None:
                continue

            with translation.override(code):
                siblings_slugs = [sibling.slug for sibling in siblings]
                if slug_value in siblings_slugs:
                    self.add_error(
                        field_name,
                        forms.ValidationError(_("This slug is already in use")),
                    )

        return cleaned_data
```

and then:
```python
class MyPage(Page):
    ...
    base_form_class = WagtailFixedAdminPageForm
```

**Comment by DiogoMarques29** [comment]
Hi @simonbru 
This was fixed here a4003f53e7848fd1318c6bd59d25921f25c6c6f3

**Comment by simonbru** [comment]
Cool, thank you !
